src/box/replication.cc

```diff
@@ -374,15 +374,6 @@ replica_set_id(struct replica *replica, uint32_t replica_id)
 		assert(instance_id == REPLICA_ID_NIL);
 		instance_id = replica_id;
 		box_broadcast_id();
-	} else if (replica->anon) {
-		/*
-		 * Set replica gc on its transition from
-		 * anonymous to a normal one.
-		 */
-		assert(replica->gc == NULL);
-		replica->gc = gc_consumer_register(
-			instance_vclock, GC_CONSUMER_REPLICA,
-			&replica->uuid);
 	}
```

Comment:
The code is actually [uncovered](https://coveralls.io/builds/70710293/source?filename=src%2Fbox%2Freplication.cc#L382). After my patch the branch is started to be taken sometimes and it crashes Tarantool since we already create a consumer in `box_process_register`.
Probably, I should drop this snippet in a separate commit.

Reply:
Done - the snippet is dropped in a separate commit.